app/src/main/java/com/example/travelapp/BinaryMinHeap.java

```diff
@@ -0,0 +1,134 @@
+package com.example.travelapp;
+
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+
+public class BinaryMinHeap<Key extends Comparable<Key>, V> {
+    private ArrayList<Entry<Key, V>> a;
+    private HashMap<V, Integer> indices;
+
+    public BinaryMinHeap() {
+        a = new ArrayList<Entry<Key, V>>();
+        indices = new HashMap<V, Integer>();
+    }
+
+    public int size() {
+        return a.size();
+    }
+
+    public boolean containsValue(V value) {
+        return indices.containsKey(value);
+    }
+
+    public void add(Key key, V value) {
+        if (key == null) {
+            throw new IllegalArgumentException("key is null!");
+        }
+        if (indices.containsKey(value)) {
+            throw new IllegalArgumentException("value is already in the min-heap!");
+        }
+
+        a.add(new Entry<Key, V>(key, value));
+        int index = a.size() - 1;
+        int parent = (index + 1) / 2 - 1;
```

Comment:
from this line alone, i'm assuming you're making the min heap as a complete binary tree. if that's the case, please comment the code to make it easier to know what's going on.